module/signature/signer_indices.go

```diff
@@ -0,0 +1,289 @@
+	signerIndices = bitutils.MakeBitVector(len(canonicalIdentifiers))
+	sigTypes = bitutils.MakeBitVector(len(stakingSigners) + len(beaconSigners))
+	signerCounter := 0
+	for canonicalIdx, member := range canonicalIdentifiers {
+		if _, ok := stakingSignersLookup[member]; ok {
+			bitutils.SetBit(signerIndices, canonicalIdx)
+			// The default value for sigTypes is bit zero, which corresponds to a staking sig.
+			// Hence, we don't have to change anything here.
+			delete(stakingSignersLookup, member)
+			signerCounter++
+			continue
+		}
+		if _, ok := beaconSignersLookup[member]; ok {
+			bitutils.SetBit(signerIndices, canonicalIdx)
+			bitutils.SetBit(sigTypes, signerCounter)
+			delete(beaconSignersLookup, member)
+			signerCounter++
+			continue
+		}
+	}
+
+	if len(stakingSignersLookup) > 0 {
+		return nil, nil, fmt.Errorf("unknown staking signers %v", stakingSignersLookup)
+	}
+	if len(beaconSignersLookup) > 0 {
+		return nil, nil, fmt.Errorf("unknown or duplicated beacon signers %v", beaconSignersLookup)
+	}
+
+	return signerIndices, sigTypes, nil
+}
+
+// DecodeSigTypeToStakingAndBeaconSigners decodes the bit-vector `sigType` to the set of
+// staking signer identities (`stakingSigners`) and the set of beacon signer identities (`beaconSigners`).
+// Prerequisite:
+//  * The input `signers` must be the set of signers in their canonical order.
+//
+// Expected Error returns during normal operations:
+//  * IncompatibleIndexSliceError indicates that `signerIndices` has the wrong length
+//  * IllegallyPaddedIndexSliceError is the vector is padded with bits other than 0
+func DecodeSigTypeToStakingAndBeaconSigners(
+	signers flow.IdentityList,
+	sigType []byte,
+) (stakingSigners flow.IdentityList, beaconSigners flow.IdentityList, err error) {
+	numberSigners := len(signers)
+	if e := validPadding(sigType, numberSigners); e != nil {
+		return nil, nil, fmt.Errorf("sigType is invalid: %w", e)
+	}
+
+	// decode bits to Identities
+	stakingSigners = make(flow.IdentityList, 0, numberSigners)
+	beaconSigners = make(flow.IdentityList, 0, numberSigners)
+	for i, signer := range signers {
+		if bitutils.ReadBit(sigType, i) == 0 {
+			stakingSigners = append(stakingSigners, signer)
+		} else {
+			beaconSigners = append(beaconSigners, signer)
+		}
+	}
+	return stakingSigners, beaconSigners, nil
+}
+
+// EncodeSignersToIndices encodes the given signerIDs into compacted bit vector.
+// PREREQUISITES:
+//  * The input `canonicalIdentifiers` must exhaustively list the set of authorized signers in their canonical order.
+//  * The input `signerIDs` represents a set, i.e. it should not contain any duplicates.
+//  * `signerIDs` must be a subset of `canonicalIdentifiers`
+//
+// RETURN VALUE:
+//  *  `signerIndices` is a bit vector. Let signerIndices[i] denote the ith bit of `signerIndices`.
+//                             ┌ 1 if and only if canonicalIdentifiers[i] is in `stakingSigners` or  `beaconSigners`
```

Comment:
I think the original comment is more accurate.

Reply:
You're right, my original suggestion reversed the order in addition to changing `stakingSigners/beaconSigners -> signerIDs`. Sorry about that. 

My suggestion here is only to use `signerIDs` in place of `stakingSigners/beaconSigners`, because `stakingSigners/beaconSigners` are not defined for this function.

Reply:
I see. Good point 👍